File: experiments/engine-kt/adapter-anthropic-messages/src/main/kotlin/com/garive/eng/kt/anthropic/Wire.kt
package com.garive.eng.kt.anthropic

/** Internal Messages HTTP vocabulary. */
internal object HttpWire {
    const val METHOD_POST: String = "POST"
    const val HEADER_CONTENT_TYPE: String = "content-type"
    const val HEADER_ACCEPT: String = "accept"
    const val MEDIA_JSON: String = "application/json"
    const val MEDIA_SSE: String = "text/event-stream"
    const val MEDIA_PDF: String = "application/pdf"
    const val MEDIA_TEXT: String = "text/plain"
}

/** Internal Messages JSON field vocabulary used across codecs. */
internal object MessageFields {
    const val TYPE: String = "type"
    const val ERROR: String = "error"
    const val DELTA: String = "delta"
    const val CONTENT_BLOCK: String = "content_block"
    const val PARTIAL_JSON: String = "partial_json"

    val CREATE: Set<String> = setOf(
        "model", "max_tokens", "messages", "stream", "system", "stop_sequences",
        "temperature", "top_p", "top_k", "tools", "tool_choice", "output_config",
        "thinking", "metadata",
    )
}

/** Internal Messages discriminators shared by parsing and lifecycle checks. */
internal object MessageKinds {
    const val MESSAGE: String = "message"
    const val ERROR: String = "error"
    const val ASSISTANT: String = "assistant"
    const val TEXT: String = "text"
    const val IMAGE: String = "image"
    const val DOCUMENT: String = "document"
    const val THINKING: String = "thinking"
    const val REDACTED_THINKING: String = "redacted_thinking"
    const val TOOL_USE: String = "tool_use"
    const val TOOL_RESULT: String = "tool_result"
    const val BASE64: String = "base64"
    const val URL: String = "url"
    const val CONTENT: String = "content"
    const val EPHEMERAL: String = "ephemeral"
    const val JSON_SCHEMA: String = "json_schema"
    const val AUTO: String = "auto"
    const val ANY: String = "any"
    const val TOOL: String = "tool"
    const val NONE: String = "none"
    const val DISABLED: String = "disabled"
    const val ENABLED: String = "enabled"
    const val ADAPTIVE: String = "adaptive"
}

File: adapters/anthropic-messages/src/wire.rs
//! Internal wire vocabulary shared by the Messages codecs.

pub(crate) const METHOD_POST: &str = "POST";
pub(crate) const HEADER_CONTENT_TYPE: &str = "content-type";
pub(crate) const HEADER_ACCEPT: &str = "accept";
pub(crate) const MEDIA_JSON: &str = "application/json";
pub(crate) const MEDIA_SSE: &str = "text/event-stream";

pub(crate) const FIELD_TYPE: &str = "type";
pub(crate) const FIELD_DELTA: &str = "delta";
pub(crate) const FIELD_CONTENT_BLOCK: &str = "content_block";
pub(crate) const FIELD_TEXT: &str = "text";
pub(crate) const FIELD_PARTIAL_JSON: &str = "partial_json";
pub(crate) const FIELD_THINKING: &str = "thinking";
pub(crate) const FIELD_SIGNATURE: &str = "signature";
pub(crate) const FIELD_CITATION: &str = "citation";

pub(crate) const KIND_ERROR: &str = "error";
pub(crate) const KIND_TEXT: &str = "text";
pub(crate) const KIND_THINKING: &str = "thinking";
pub(crate) const KIND_REDACTED_THINKING: &str = "redacted_thinking";
pub(crate) const KIND_TOOL_USE: &str = "tool_use";

pub(crate) const EVENT_MESSAGE_START: &str = "message_start";
pub(crate) const EVENT_CONTENT_BLOCK_START: &str = "content_block_start";
pub(crate) const EVENT_CONTENT_BLOCK_DELTA: &str = "content_block_delta";
pub(crate) const EVENT_CONTENT_BLOCK_STOP: &str = "content_block_stop";
pub(crate) const EVENT_MESSAGE_DELTA: &str = "message_delta";
pub(crate) const EVENT_MESSAGE_STOP: &str = "message_stop";
pub(crate) const EVENT_PING: &str = "ping";

pub(crate) const DELTA_TEXT: &str = "text_delta";
pub(crate) const DELTA_INPUT_JSON: &str = "input_json_delta";
pub(crate) const DELTA_THINKING: &str = "thinking_delta";
pub(crate) const DELTA_SIGNATURE: &str = "signature_delta";
pub(crate) const DELTA_CITATIONS: &str = "citations_delta";

pub(crate) const CREATE_FIELDS: &[&str] = &[
    "model",
    "max_tokens",
    "messages",
    "stream",
    "system",
    "stop_sequences",
    "temperature",
    "top_p",
    "top_k",
    "tools",
    "tool_choice",
    "output_config",
    "thinking",
    "metadata",
];
